internal/ingress/controller/parser/parser_test.go

```diff
@@ -1732,6 +1732,175 @@ func TestKongProcessClasslessIngress(t *testing.T) {
 
 func TestKnativeIngressAndPlugins(t *testing.T) {
 	assert := assert.New(t)
+	t.Run("knative ingress annotated with konghq.com/override", func(t *testing.T) {
+		ingresses := []*knative.Ingress{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "knative-ingress-with-override",
+					Namespace: "foo-ns",
+					Annotations: map[string]string{
+						"networking.knative.dev/ingress.class":                      annotations.DefaultIngressClass,
+						annotations.AnnotationPrefix + annotations.ConfigurationKey: "https-only",
+					},
+				},
+				Spec: knative.IngressSpec{
+					Rules: []knative.IngressRule{
+						{
+							Hosts: []string{"my-func.example.com"},
+							HTTP: &knative.HTTPIngressRuleValue{
+								Paths: []knative.HTTPIngressPath{
+									{
+										Path: "/",
+										AppendHeaders: map[string]string{
+											"foo": "bar",
+										},
+										Splits: []knative.IngressBackendSplit{
+											{
+												IngressBackend: knative.IngressBackend{
+													ServiceNamespace: "foo-ns",
+													ServiceName:      "foo-svc",
+													ServicePort:      intstr.FromInt(42),
+												},
+												Percent: 100,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		kongIngresses := []*configurationv1.KongIngress{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "https-only",
+					Namespace: "foo-ns",
+					Annotations: map[string]string{
+						"networking.knative.dev/ingress.class": annotations.DefaultIngressClass,
+					},
```

Comment:
Why is this annotation added in here?
This annotation should only be interpreted on the Knative Ingress resource.

Reply:
Thanks for pointing it out. I probably copy-pasted from another section and forgot to remove it.